samples/shields/x_nucleo_iks01a3/sensorhub/sample.yaml

```diff
@@ -2,8 +2,7 @@ sample:
   name: X-NUCLEO-IKS01A3 sensor shield
 tests:
   test:
-    platform_exclude: disco_l475_iot1
     harness: shield
     tags: shield
-    depends_on: arduino_i2c arduino_header
-    platform_exclude: disco_l475_iot1
+    depends_on: arduino_i2c arduino_gpio
+    platform_exclude: disco_l475_iot1 stm32mp157c_dk2
```

Comment:
Thinking again about it, I think we could also add a stm32mp157c_dk2 dedicated overlay in the shield removing the pins that don't match, rather than excl
This would allow to keep using the board wit the shield (which totally works).
What do you think?

Reply:
Agreed, but more work needs to be done for that, let's address that after we get this merged to fix the build issues.

See but #19078

Reply:
I tested successfully the following:
`boards/shields/x_nucleo_iks01a2/boards/stm32mp157c_dk2.overlay`
```
&arduino_i2c {
	lsm303agr-magn@1e {
		/delete-property/ irq-gpios;	/* A3 */
	};

	lsm303agr-accel@19 {
		/delete-property/ irq-gpios;	/* A3 */
	};
};
```